<?xml version="1.0" ?>

<robot name="m1_workcell">

    <group name="robot_only">
        <!-- <chain base_link="positioner_tool0" tip_link="tool0" /> -->
        <chain base_link="base_link" tip_link="tool0" />
    </group>
    
    <group name="manipulator">
        <chain base_link="positioner_tool0" tip_link="tool0" />
        <!-- <chain base_link="world" tip_link="tool0" /> -->
    </group>


    <!-- Groups kinematic plugins -->
    <kinematics_plugin_config filename="package://abb_example/config/m2_kin.yaml"/>

    <!--Contact Managers plugins -->
    <contact_managers_plugin_config filename="package://tesseract_ros_examples/config/contact_manager_plugins.yaml"/>

    <!--inverse kinematics-->
    <group name="positioner_only">
        <chain base_link="positioner_base_link" tip_link="positioner_tool0"/>
    </group>

    <!-- <group_opw group="manipulator" a1="0.15" a2="-0.115" b="0" c1="0.445" c2="0.7" c3="0.795" c4="0.085" offsets="0.000000 0.000000 -1.570796 0.000000 0.000000 0.000000" sign_corrections="1 1 1 1 1 1"/> -->

    <!--GROUP ROBOT WITH EXTERNAL POSITIONER INVERSE KINEMATICS PARAMETERS-->
    <!-- <group_rep group="manipulator">
      <manipulator group="robot" ik_solver="OPWInvKin" reach="2.3"/>
      <positioner group="positioner_only" fk_solver="KDLFwdKinChain">
        <joint name="positioner_joint_1" resolution="0.1"/>
        <joint name="positioner_base_joint" resolution="0.1"/>
      </positioner>
    </group_rep> -->



    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
    <disable_collisions link1="base_link" link2="link_1" reason="Adjacent" />
    <disable_collisions link1="base_link" link2="link_2" reason="Never" />
    <disable_collisions link1="base_link" link2="link_3" reason="Never" />
    <disable_collisions link1="base_link" link2="link_4" reason="Never" />
    <!-- <disable_collisions link1="base_link" link2="positioner_link_1" reason="Never" />
    <disable_collisions link1="base_link" link2="positioner_link_2" reason="Never" />
    <disable_collisions link1="base_link" link2="positioner_tool0" reason="Never" /> -->

    <disable_collisions link1="link_1" link2="link_2" reason="Adjacent" />
    <disable_collisions link1="link_1" link2="link_3" reason="Never" />
    <disable_collisions link1="link_1" link2="link_4" reason="Never" />
    <disable_collisions link1="link_1" link2="link_5" reason="Never" />
    <disable_collisions link1="link_1" link2="link_6" reason="Never" />

    <disable_collisions link1="link_2" link2="link_3" reason="Adjacent" />
    <disable_collisions link1="link_2" link2="link_4" reason="Never" />
    <disable_collisions link1="link_2" link2="link_5" reason="Never" />
    <disable_collisions link1="link_2" link2="link_6" reason="Never" />
    <disable_collisions link1="link_3" link2="link_4" reason="Adjacent" />
    <disable_collisions link1="link_3" link2="link_5" reason="Never" />
    <disable_collisions link1="link_3" link2="link_6" reason="Never" />
    <disable_collisions link1="link_4" link2="link_5" reason="Adjacent" />
    <disable_collisions link1="link_4" link2="link_6" reason="Never" />
    <disable_collisions link1="link_5" link2="link_6" reason="Adjacent" />

    <!-- <disable_collisions link1="positioner_base_link" link2="positioner_link_1" reason="Adjacent" />
    <disable_collisions link1="positioner_base_link" link2="positioner_link_2" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="positioner_tool0" reason="Never" />

    <disable_collisions link1="positioner_base_link" link2="base_link" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="link_1" reason="Never" /> -->
    <!--disable_collisions link1="positioner_base_link" link2="link_2" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="link_3" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="link_4" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="link_5" reason="Never" />
    <disable_collisions link1="positioner_base_link" link2="link_6" reason="Never" /-->

    <!-- <disable_collisions link1="positioner_link_1" link2="positioner_link_2" reason="Adjacent" />
    <disable_collisions link1="positioner_link_1" link2="positioner_tool0" reason="Never" />
    <disable_collisions link1="positioner_link_1" link2="link_1" reason="Never" /> -->
    <!--disable_collisions link1="positioner_link_1" link2="link_2" reason="Never" />
    <disable_collisions link1="positioner_link_1" link2="link_3" reason="Never" />
    <disable_collisions link1="positioner_link_1" link2="link_4" reason="Never" />
    <disable_collisions link1="positioner_link_1" link2="link_5" reason="Never" />
    <disable_collisions link1="positioner_link_1" link2="link_6" reason="Never" /-->

    <!-- <disable_collisions link1="positioner_link_2" link2="positioner_tool0" reason="Adjacent" />
    <disable_collisions link1="positioner_link_2" link2="link_1" reason="Never" /> -->
    <!--disable_collisions link1="positioner_link_2" link2="link_2" reason="Never" />
    <disable_collisions link1="positioner_link_2" link2="link_3" reason="Never" />
    <disable_collisions link1="positioner_link_2" link2="link_4" reason="Never" />
    <disable_collisions link1="positioner_link_2" link2="link_5" reason="Never" />
    <disable_collisions link1="positioner_link_2" link2="link_6" reason="Never" /-->

</robot>
